source/smeagle.cpp

```diff
@@ -21,6 +21,28 @@ using namespace smeagle;
 
 Smeagle::Smeagle(std::string _library) : library(std::move(_library)) {}
 
+// Determine if the library has exceptions with smeagle
+bool Smeagle::has_exceptions() {
+  Symtab *obj = NULL;
+  std::vector<ExceptionBlock *> exceptions;
+
+  // Read the library into the Symtab object, cut out early if there's error
+  if (not Symtab::openFile(obj, library)) {
```

Comment:
`not` is technically part of C++, but it usually doesn't get much use as it's the spelled-out version of `!`. It makes it look more Pythonic, though! :D

Reply:
It would also be nice if we didn't have to re-parse the binary between calling this and `parse`. It's probably not a big deal right now, but if we have multi-gigabyte files, it could get noticeable.

Reply:
I'll remove the extra command!

I totally agree on not wanting to load twice. Right now I'm using them exclusively (e.g., not parsing AND running that function) but we can fix if/when we decide loading once is necessary).

Reply:
That makes sense. Even running this on E4S wouldn't be a problem. The binaries there are fairly small. It's just something to put in the back of our minds for now.

Reply:
I already ran it on the build-sandbox cache! Is e4s in the same AWS space?

Reply:
I don't see it - we can ask on Friyay.

Reply:
I think there is a copy of E4S somewhere in AWS, but I'm not sure. The official distribution is at e4s.io.